### ISSUE TITLE
Enable Link-Time Optimization (LTO) once again

### ISSUE THREAD

**Issue by zamazan4ik** [Amjad50/plastic#10]
Hi!

I noticed that in the `Cargo.toml` [file](https://github.com/Amjad50/plastic/blob/master/Cargo.toml) Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where additionally to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Using ThinLTO should also help to reduce the build-time overhead with LTO. If we enable it on the Cargo profile level, users, who install the application with `cargo install`, will get the LTO-optimized version "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

I see that LTO was [disabled](https://github.com/Amjad50/plastic/commit/fe271ee10aa80c92b91f3f9f7967e56aa0ae4197) 4 years ago due to some build issues. I think after such a period of time we can reconsider enabling this compiler option. I made some quick tests on my local machine (Fedora 40, AMD Ryzen 9 5900x): `lto = true` reduces the `plastic` binary size from 17 Mib to 13 Mib (I didn't perform a performance comparison though) - quite a good result!

Thank you.

**Comment by Amjad50** [comment]
Thanks for the suggestion you are correct.
